server/index.js

```diff
@@ -0,0 +1,155 @@
+const express = require('express');
+const app = express();
+
+const PORT = process.env.PORT || 8000;
+
+// const knex = require('./middleware/knex');
+const APP_MODE = 'development';
+const Knex = require('knex');
+const dbConfig = require('./knexfile');
+const knex = Knex(dbConfig[APP_MODE]);
+
+const bcrypt = require('bcrypt-as-promised');
+// const cookieSession = require('cookie-session');
+
+const bodyParser = require('body-parser');
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: false}));
+
+const request = require('request');
+const cheerio = require('cheerio');
+
+// app.use(cookieSession({
+//   name: 'auth_app_user',
+//   keys: ['socure1', 'socure2']
+// }));
+
+//CORS for Client
+app.options('/*', (req, res, next) => {
+  console.log(req.method, req.headers)
+  res.set('Access-Control-Allow-Origin', '*');
+  res.set("Access-Control-Allow-Headers", "Content-Type, authorization");
+  res.set("Access-Control-Allow-Credentials", "true");
+  res.set('Access-Control-Allow-Methods', 'GET, POST, OPTIONS, PUT, PATCH');
+
+  res.end();
+});
+
+//ROUTING
+
+  //ADD USER
+app.post('/api/users/', function (req, res) {
+
+  bcrypt.hash(req.body.password, 12)
+    .then(function (digest) {
+      // use knex to insert the digested password
+      knex('users')
+      .insert({
+          email: req.body.email,
+          hashed_password: digest
+        }, '*')
+        .then(function (users) {
+           console.log(users)
+          const user = users[0];
+
+          res.send(`Welcome ${user.email}`)
+        })
+    })
+
+    .catch(function (err) {
+      console.log(err);
+      res.send("Error");
+    });
+
+});
+
+  //MAKE POSTS
+app.use('/api', require('./routes/posts'))
+
+  //INTERPRET (SCRAPER)
+app.get('/test', function(req, res) {
+
+  var str = "Macaroni"
+
+  var splitStr = str.split(" ");
+
+  splitStr.forEach(function(el){
+    el.replace(/\.$/, " ");
+  })
+
+  var newStr = splitStr.filter(function (el){
+    return el !== 'A'
```

Comment:
Moving this into an array and using `indexOf` or using a RegExp could help clean this up